data_model/src/lib.rs

```diff
@@ -397,6 +398,24 @@ impl IdentifiableBox {
     }
 }
 
+impl<'live> TryFrom<&'live IdentifiableBox> for &'live dyn HasMetadata {
```

Comment:
```suggestion
impl<'idbox> TryFrom<&'idbox IdentifiableBox> for &'idbox dyn HasMetadata {
```

Reply:
Are the lifetimes absolutely necessary here?

Reply:
yes, it's necessary

Reply:
while they are needed in trait definition, they are most likely not need on the method implementation below